app/src/main/java/com/example/newsportal/app/topnews/NewsViewModel.kt

```diff
@@ -1,19 +1,26 @@
 package com.example.newsportal.app.topnews
 
+import android.util.Log
 import androidx.lifecycle.*
 import androidx.lifecycle.ViewModel
 import com.example.newsportal.data.NewsRepository
+import com.example.newsportal.domain.model.Article
+import com.example.newsportal.domain.usecases.GetNewsUseCase
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
 class NewsViewModel (
-    private val mRepository: NewsRepository
+    private val getNewsUseCase: GetNewsUseCase
 ): ViewModel() {
 
-    val data = mRepository.data
+    val data = MutableLiveData<List<Article>>()

```

Comment:
You shouldn't expose mutable live data to clients, make private it private and add public property that will expose LiveData
private val _data = MutableLiveData<List<Article>>()
val data: LiveData<List<Article>> = _data